### PR TITLE
scripts: tests: Blackbox test expansion - coverage

### PR DESCRIPTION
**Pull request by KamilxPaszkiet**
Adds tests related to coverage flags:
`--coverage-tool`
`--coverage-basedir`
`--coverage-formats`

Move to coverage.py:
`--coverage`
`--enable-coverage`

### CODE REVIEW

File: scripts/tests/twister_blackbox/test_coverage.py [hunk 1]
```diff
@@ -0,0 +1,294 @@
+#!/usr/bin/env python3
+# Copyright (c) 2023 Intel Corporation
```
Comment (LukaszMrugala) [inline scripts/tests/twister_blackbox/test_coverage.py:2]: 2024, not 2023

File: scripts/tests/twister_blackbox/test_coverage.py [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+#
+# SPDX-License-Identifier: Apache-2.0
+"""
+Blackbox tests for twister's command line functions
+"""
+
+import importlib
+import re
+
```
Comment (LukaszMrugala) [inline scripts/tests/twister_blackbox/test_coverage.py:11]: Remove blank line

File: scripts/tests/twister_blackbox/test_coverage.py [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+        out, err = capfd.readouterr()
+        sys.stdout.write(out)
+        sys.stderr.write(err)
+
+        for f_name in file_name:
+            path = os.path.join(out_path, f_name)
+            assert os.path.exists(path), f'file not found {f_name}'
+
+        assert str(sys_exit.value) == '0'
```
Comment (LukaszMrugala) [inline scripts/tests/twister_blackbox/test_coverage.py:168]: Best to check the exit value in the first assert (if that breaks, you know more about the error).

File: scripts/tests/twister_blackbox/test_coverage.py [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+
+    @pytest.mark.parametrize(
+        'test_path, test_platforms, file_name',
+        TESTDATA_1,
+        ids=[
+            'coverage',
+        ]
+    )
+    def test_coverage(self, capfd, test_path, test_platforms, out_path, file_name):
+        args = ['-i','--outdir', out_path, '-T', test_path, '--coverage', '--coverage-tool', 'gcovr'] + \
```
Comment (LukaszMrugala) [inline scripts/tests/twister_blackbox/test_coverage.py:151]: I like to split args into separate lists, to have a clear picture of what are the flags under tests.

**Review by LukaszMrugala — CHANGES_REQUESTED** [review]

File: scripts/tests/twister_blackbox/test_coverage.py (hunk 4 above)
Comment (KamilxPaszkiet) [reply scripts/tests/twister_blackbox/test_coverage.py]: done, now it's more clear

**Review by LukaszMrugala — DISMISSED** [review]

**Review by LukaszMrugala — DISMISSED** [review]

**Review by nashif — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
@KamilxPaszkiet this needs a rebase

**Review by nashif — APPROVED** [review]

**Review by LukaszMrugala — APPROVED** [review]

**Pull request merged**